cli/dcoscli/main.py

```diff
@@ -67,17 +64,31 @@ def _main():
             msg = 'Unable to get the cluster_id of the cluster.'
             logger.exception(msg)
 
+    # send args call to segment.io
+    reporting_executor = ThreadPoolExecutor(max_workers=2)
```

Comment:
Use `with ThreadPoolExecutor` so that it gets shutdown and more importantly we join/wait on all of the tasks finishing.